### PR TITLE
bugfix: Snippet generator fails: No enum constant org.jenkinsci.plugins.pipeline.maven.MavenPublisherStrategy (JENKINS-71073)

### PR DESCRIPTION
**Pull request by bguerin**
https://issues.jenkins.io/browse/JENKINS-71073

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by jglick — CHANGES_REQUESTED** [review]
I think it is https://javadoc.jenkins.io/hudson/model/Item.html#EXTENDED_READ you were looking for. This is what would be granted on a branch project (for example) when `CONFIGURE` is not and you click the **View Configuration** link. Merely checking `READ` is not safe in general as it could allow someone is not permitted to look at project configurations (but only see logs, etc.) to obtain lists of various global configuration objects, which could be a starting point for an attack if not inherently dangerous.

@daniel-beck does our custom security scanner check for this sort of thing, or simply a _missing_ ACL check?

**Comment by daniel-beck** [comment]
> does our custom security scanner check for this sort of thing, or simply a _missing_ ACL check?

Missing check, as well as `#hasPermission` having its return value ignored.

What specifically would you check here? I guess the way `#getItemByFullName` etc. check for Item/Read already, this check is one that's unlikely to be useful?

File: jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStep.java [hunk 1]
```diff
@@ -78,7 +76,7 @@ public class WithMavenStep extends Step {
     private String mavenLocalRepo = "";
     private List<MavenPublisher> options = new ArrayList<>();
     private MavenPublisherStrategy publisherStrategy = MavenPublisherStrategy.IMPLICIT;
-    private boolean traceability = true;
+    private boolean traceability = false;
```
Comment (jglick) [inline jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStep.java:79]: Did you accidentally mix b688eb87db0a9e85b6c0364e874912f8310de8ee into an unrelated PR?

Comment (bguerin) [reply jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStep.java]: I needed a fix done on my other open PR for test to pass.

I have extracted this fix to a dedicated commit on master, and rebased my two PR on top of it.

Should be fine now

**Comment by bguerin** [comment]
Dear Jenkins experts, @daniel-beck and @jglick 

While I understand, and agree, that security is a major point to take care of, I do not understand your dialog.
Is there anything I should do more on my side ?

**Review by jglick — APPROVED** [review]
Looks fine to me. I have not tried it to confirm.

**Comment by bguerin** [comment]
Ok, thanks @jglick 

I have tested my changes and confirmed they fixed the problem you describe in the JIRA issue

**Pull request merged**